### DISCUSSION TITLE
Improving Workflow Speed

### DISCUSSION THREAD

**Discussion by Patrick-Galapon** [StackStorm/st2#5982] [Workflows]
Hello, 
Looking to improve performance speed on my stackstorm workflow.  The workflow takes about 30s to complete based on the execution historyand is comprised of the following:
- retrieves inputs message body, slack channel, timestamp key and emoji names from a previous actions
- retrieves slack api key, icon image and message timestamp (using timestamp key) from the datastore
- sends slack message based on the above data given.  If a message timestamp was retrieved, the slack message will reply to the message tied to the timestamp
- adds the emojis
- stores message timestamp into the datastore for future use (in case this message will be replied to)

Any insight or feedback in ways I can reduce the performance speed of 30s is appreciated.

**Comment by guzzijones** [comment]
pack it all up in one python script to avoid mongo writes and reads.
